spec/acceptance/certs_tar_extract_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'spec_helper_acceptance'
+
+describe 'certs with tar archive' do
+  before(:context) do
+    apply_manifest('include certs', catch_failures: true)
+
+    pp = <<-PUPPET
+      class { 'certs':
+        generate => true,
+        deploy   => false,
+      }
+
+      class { 'certs::foreman_proxy_content':
+        foreman_proxy_fqdn => 'foreman-proxy.example.com',
+        certs_tar          => '/root/foreman-proxy.example.com.tar.gz',
+      }
+    PUPPET
+
+    apply_manifest(pp, catch_failures: true)
+    on default, "rm -rf /root/ssl-build"
+
+    install_certs = <<-PUPPET
+      class { 'certs':
+        tar_file  => '/root/foreman-proxy.example.com.tar.gz',
+        generate  => false,
+        node_fqdn => 'foreman-proxy.example.com',
+      }
+
+      include certs::apache
+    PUPPET
+
+    apply_manifest(install_certs, catch_failures: true)
```

Comment:
Can we also check for idempotency?
```suggestion
    apply_manifest(install_certs, catch_failures: true)
    apply_manifest(install_certs, catch_changes: true)
```

Reply:
I can confirm that this fails idempotency which is why I have it written the way I do. These tests will allow us to try and fix that as we'll have something testing our baseline. The reason for failure is due to the fact that is always extracts the tarball or always creates it without any care if it already exists or not.

Reply:
@ehelms with the understanding that this PR only adds testing, and the broken idempotence is already present in the module, what are the consequences for the foreman-installer if the module always creates or extracts the tarball?

Reply:
I think that is the expected behavior in the installer right now as it recreates it fresh every time it runs. I think making this idempotent will require some thought as to how that applies to the installer given the current behavior.

Reply:
Perhaps an explicit comment that it isn't idempotent makes sense?

Reply:
Comment added